clang-tools-extra/test/clang-tidy/checkers/bsl-using-ident-unique-namespace.cpp

```diff
@@ -0,0 +1,81 @@
+// RUN: %check_clang_tidy %s bsl-using-ident-unique-namespace %t
+// #include <cstdint>
+
+using func = void (*)(int, int);	// Compliant
+
+namespace n1
+{
+	using func = void (*)(int, int);
```

Comment:
func here is non-compliant because ::func and n1::func have the same name. n1::func and n2::func would be ok, but if an alias is defined in the global namespace, or in a parent namespace, any child namespace with that same name would not be compliant. Think of a tree structure. So long as a child never has the same name as a parent, you are good.